### PR TITLE
[DB+API]:  [DB]: added user to users table, and users-roles to users_roles table [API] Added AuthenticationController class

### PR DESCRIPTION
**Pull request by RostyslavKuzmych**

### CODE REVIEW

File: src/main/java/application/security/CustomUserDetailsService.java [hunk 1]
```diff
@@ -16,7 +16,7 @@ public class CustomUserDetailsService implements UserDetailsService {
 
     @Override
     public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
-        return userRepository.findByEmail(email).orElseThrow(()
+        return userRepository.findUserByEmail(email).orElseThrow(()
                 -> new EntityNotFoundException(EXCEPTION + email));
```
Comment (DEz-1996) [inline src/main/java/application/security/CustomUserDetailsService.java:20]: rename constant EXCEPTION to CANT_FIND_BY_EMAIL

File: src/main/java/application/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public class UserServiceImpl implements UserService {
+    private static final Long USER_ID = 2L;
+    private static final String EXCEPTION = "User with email %s is already authenticated";
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
+
+    @Override
+    public UserResponseRegistrationDto register(UserRequestRegistrationDto
+                                                            userRequestRegistrationDto) {
```
Comment (DEz-1996) [inline src/main/java/application/service/impl/UserServiceImpl.java:28]:
```suggestion
    public UserResponseRegistrationDto register(
            UserRequestRegistrationDto userRequestRegistrationDto) {

```

File: src/main/java/application/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private static final Long USER_ID = 2L;
+    private static final String EXCEPTION = "User with email %s is already authenticated";
```
Comment (DEz-1996) [inline src/main/java/application/service/impl/UserServiceImpl.java:21]:
```suggestion
    private static final String EMAIL_RESERVED= "User with email %s already exists";

```

File: src/main/java/application/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (DEz-1996) [inline src/main/java/application/service/impl/UserServiceImpl.java:28]:
```suggestion
    public UserResponseRegistrationDto register(
            UserRequestRegistrationDto userRequestRegistrationDto) {

```

File: src/main/resources/db/changelog/changes/11-add-user-to-users-table.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  - changeSet:
+      id: add_user_to_users_table
+      author: rostyslav_kuzmych
+      changes:
+        - insert:
+            tableName: users
+            columns:
+              - column:
+                  name: id
+                  value: 1
```
Comment (DEz-1996) [inline src/main/resources/db/changelog/changes/11-add-user-to-users-table.yaml:11]:
```suggestion

```

File: src/main/resources/db/changelog/changes/12-add-users_roles-to-users_roles-table.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  - changeSet:
+      id: add_users_roles_to_users_roles_table
+      author: rostyslav_kuzmych
+      changes:
+        - insert:
+            tableName: users_roles
+            columns:
+              - column:
+                  name: user_id
+                  value: 1
```
Comment (DEz-1996) [inline src/main/resources/db/changelog/changes/12-add-users_roles-to-users_roles-table.yaml:11]:
```suggestion
                  valueComputed: (SELECT id FROM users WHERE email='alice@gmail.com')

```

File: src/main/resources/db/changelog/changes/12-add-users_roles-to-users_roles-table.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+                  value: 2
+        - insert:
+            tableName: users_roles
+            columns:
+              - column:
+                  name: user_id
+                  value: 1
+              - column:
+                  name: role_id
+                  value: 1
```
Comment (DEz-1996) [inline src/main/resources/db/changelog/changes/12-add-users_roles-to-users_roles-table.yaml:23]:
```suggestion
                  valueComputed: (SELECT id FROM roles WHERE name='ADMIN')

```

**Review by DEz-1996 — CHANGES_REQUESTED** [review]
Some fixes are needed

File: src/main/java/application/model/User.java [hunk 7]
```diff
@@ -29,6 +32,7 @@ public class User implements UserDetails {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
+    @Getter
```
Comment (AgroFix) [inline src/main/java/application/model/User.java:35]: If I'm not mistaken, if you have @Data, you don't need @Getter

**Review by AgroFix — APPROVED** [review]

**Pull request merged**